app/domain/authentication/authn_jwt/jwt_initial_validate_input.rb

```diff
@@ -0,0 +1,56 @@
+require 'command_class'
+
+# This class
+module Authentication
+  module AuthnJwt
+
+    JWTInitialValidateInput ||= CommandClass.new(
+      dependencies: {
+        # ValidateWebserviceIsWhitelisted is also calling ValidateAccountExists
+        # it's better to be explicit and on the safe side
+        validate_account_exists: ::Authentication::Security::ValidateAccountExists.new,
+        validate_webservice_is_whitelisted: ::Authentication::Security::ValidateWebserviceIsWhitelisted.new,
+        jwt_validate_body: Authentication::AuthnJwt::JWTValidateRequestBody.new
+      },
+      inputs: %i[authentication_parameters enabled_authenticators]
+    ) do
+
+      def call
+        validate_account_exists
+        validate_webservice_is_whitelisted
+        jwt_validate_body
+      end
+
+      private
+
+      def validate_account_exists
+        @validate_account_exists.(
+          account: @authentication_parameters.account
+        )
+      end
+
+      def validate_webservice_is_whitelisted
+        @validate_webservice_is_whitelisted.(
+            webservice: webservice,
+            account: @authentication_parameters.account,
+            enabled_authenticators: @enabled_authenticators
+        )
+      end
+
+      def jwt_validate_body
+        @jwt_validate_body.(
+          body_string: @authentication_parameters.request.body.read
```

Comment:
i think you can get the same value by using `@authentication_parameters.credentials`. i don't know how `@authentication_parameters` is different than `@authentication_input` but in the latter you can use `credentials` instead of reading the body again.

Reply:
btw, in `authn-oidc` the body is:
```
id_token: <JWT>
```
and in authn-gcp it is:
```
jwt: <JWT>
```

here the body is only the token itself without a key?

Reply:
@orenbm  you are right `authentication_input` cares to read body to `credentials`

Reply:
It was defined in the feature doc
![image](https://user-images.githubusercontent.com/41368629/117644266-fabd0d00-b191-11eb-8ac5-54ef65be54e0.png)

Reply:
So we have a third format with `jwt=` prefix... :-(
Let's be aligned at least with `gcp` means `jwt: ` prefix.

Reply:
@sashaCher you can look at [this file](https://github.com/cyberark/conjur/blob/2d28f6dff4297338855fd109c16b6b7baa365fdc/app/domain/authentication/jwt/decoded_credentials.rb#L6) for reference. i would use the `Jwt::DecodedCredentials` class in your case as well as it handles the same logic and will validate the request for you

Reply:
> So we have a third format with `jwt=` prefix... :-(
> Let's be aligned at least with `gcp` means `jwt: ` prefix.

@sashaCher i was wrong. it is `jwt=` for GCP as well. go ahead with this.

Reply:
@orenbm Thank you!